plugins/module_utils/session_utils.py

```diff
@@ -0,0 +1,365 @@
+            "ca_path": self.ca_path,
+            "use_proxy": self.use_proxy,
+            "headers": req_header,
+            "timeout": api_timeout,
+            "follow_redirects": 'all',
+        }
+        return url_kwargs
+
+    def _args_session(self, path, method, api_timeout, headers=None):
+        """
+        Returns a dictionary containing the arguments needed to establish a session.
+
+        :param path: A string representing the path of the API endpoint.
+        :param method: A string representing the HTTP method to be used.
+        :param api_timeout: An integer representing the timeout for the API request.
+        :param headers: An optional dictionary containing additional headers to be included in the
+        request.
+        :return: A dictionary containing the arguments needed to establish a session, including the
+        URL arguments, headers, and API timeout.
+        """
+        req_header = self._headers
+        if headers:
+            req_header.update(headers)
+        url_kwargs = self._url_common_args_spec(method, api_timeout, headers=headers)
+        return url_kwargs
+
+    def invoke_request(self, uri, method, data=None, query_param=None, headers=None, api_timeout=None, dump=True):
+        """
+        Sends an HTTP request to the specified URI using the specified method.
+
+        Args:
+            uri (str): The URI to send the request to.
+            method (str): The HTTP method to use for the request.
+            data (Optional[dict]): The data to send with the request. Defaults to None.
+            query_param (Optional[dict]): The query parameters to include in the request URL.
+            Defaults to None.
+            headers (Optional[dict]): The headers to include in the request. Defaults to None.
+            api_timeout (Optional[int]): The timeout value for the request in seconds.
+              Defaults to None.
+            dump (bool): Whether to JSON dump the data before sending the request. Defaults to True.
+
+        Returns:
+            OpenURLResponse: The response object containing the response data.
+
+        Raises:
+            HTTPError: If an HTTP error occurs during the request.
+            URLError: If a URL error occurs during the request.
+            SSLValidationError: If an SSL validation error occurs during the request.
+            ConnectionError: If a connection error occurs during the request.
+        """
+        try:
+            url_kwargs = self._args_session(uri, method, api_timeout, headers=headers)
+            if data and dump:
+                data = json.dumps(data)
+            url = self._build_url(uri, query_param=query_param)
+            resp = open_url(url, data=data, **url_kwargs)
+            resp_data = OpenURLResponse(resp)
+        except (HTTPError, URLError, SSLValidationError, ConnectionError) as err:
+            raise err
+        return resp_data
+
+    def __enter__(self):
```

Comment:
I think you can remove this if there is not implementation

Reply:
Done